Feature/smart pointer in file reader

Code7R opened the pull request.

Code7R committed:
Using smart pointers instead of naked buffers

Also extending smart pointers with more unique_ptr semantics

Code7R committed:
Convert file reading into OOP style

Code7R committed:
Added reading with timeout

Also replacing the tmpfile hack with classic pipe because it fails badly
in non-block mode.

gijsbers commented:
1. There are `if(` and `switch(`, but after a C keyword a space must follow first.
2. Some operators like `&&` and `+` are not surrounded by spaces.
3. The `if (tfd > 2) close(tfd);` was meant to avoid a filedescriptor leak.
If the child closes 2, but not the `>2`, then the reader is left hanging needlessly.
4. The `#ifndef F_GETFL` must be removed since F_GETFL must be available,
otherwise a missing header include goes undetected.

Code7R committed:
Added forgotten yfileio.h header

Code7R committed:
Cosmetics

- more spaces, more brackets over expressions in lengthy chains
- added explicit #error if F_GETFL is N/A
- added icewm code style file for Eclipse

Code7R commented:
So far, so good, but I don't agree on point (3). With this change, tfd is gone. pipe() descriptors are used. Checking for specific values there does not make sense, one has to be careful that the parent side is closed properly, though.
